lightning-invoice/src/utils.rs

```diff
@@ -267,29 +268,47 @@ where
 	// the hints across our real nodes we add one hint from each in turn until no node has any hints
 	// left (if one node has more hints than any other, these will accumulate at the end of the
 	// vector).
-	let mut invoice_hints: Vec<RouteHint> = Vec::new();
-	let mut hint_idx = 0;
+	const MAX_HINTS: usize = 3;
 
-	loop {
-		let mut remaining_hints = false;
+    zip_nested_vectors(phantom_hints, MAX_HINTS)
 
-		for hints in phantom_hints.iter() {
-			if invoice_hints.len() == 3 {
-				return invoice_hints
-			}
-
-			if hint_idx < hints.len() {
-				invoice_hints.push(hints[hint_idx].clone());
-				remaining_hints = true
-			}
-		}
-
-		if !remaining_hints {
-			return invoice_hints
-		}
+}
 
-		hint_idx +=1;
-	}
+// Zip multiple nested vectors
+// Similar to the zip method on a vector, it combines the vectors by index - zero index comes first
+// then second, etc.
```

Comment:
`zip` returns 2-tuples whereas this iterator returns single elements. I don't think we should use "zip" in the name, as that implies tupling elements from each input.

Reply:
Do you have a suggestion for what to name it?

My reasoning for having `zip` in the name is because it does zip multiple arrays together by their index.  I couldn't think of another name for that.  Like in Python, it is also called zip when combine two iterables by their index.